testcases/OpTestPstore.py

```diff
@@ -0,0 +1,176 @@
+        res = self.con.run_command( "cat /boot/config-`uname -r` | grep -i --color=never 'CONFIG_PSTORE=y'")
+        print(res)
+        if(res[0]!='CONFIG_PSTORE=y'):
+            self.fail("PSTORE is not configured")
+
+    def kernel_crash(self):
+        #1. This function will trigger a Crash
+        #2. Check for lpar or baremetal
+        try:
+            self.con.run_command( "echo c > /proc/sysrq-trigger")
+        except (KernelOOPS, KernelKdump, KernelPanic, KernelFADUMP):
+            self.cv_SYSTEM.goto_state(OpSystemState.OFF)
+            self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        res = self.con.run_command("tail /proc/cpuinfo | grep platform| awk -F : '{print $2}'")
+        if 'pSeries' in res[0].strip():
+            self.pstore_file_check(['dmesgnvram', 'rtasnvram', 'powerpcofwnvram', 'powerpccommonnvram'])
+        elif 'PowerNV' in res[0].strip():
+            self.pstore_file_check(['dmesgnvram', 'powerpcopalnvram', 'powerpccommonnvram'])
+            
+    def pstore_file_check(self, param):
+        #This function will validate new files are created
+        #or not under /sys/fs/pstore after crash
+        time_init = self.con.run_command( "date +%s")
+        result = self.con.run_command( 'ls -1 /sys/fs/pstore')
+        l = []
+        for line in result:
+            files_list =  "".join(re.findall("[a-zA-Z]",line))
+            l.append(files_list)
+        for files in param:
+            if files not in l:
+                self.fail("New %s is not created" % files)
+        for i in result:
+            self.con.run_command( "cd /sys/fs/pstore")
+            time_created = self.con.run_command( "stat -c%%Z %s" % i)
+            if (time_init > time_created):
+                print("New file %s is created" % i)
+            else:
+                print("New file %s is not created" % i)
+
+    def setup_fadump(self):
+        if self.distro == 'SLES':
+            self.con.run_command( 'sed -i \'/^KDUMP_SAVEDIR=/c\KDUMP_SAVEDIR=\/var\/crash\' /etc/sysconfig/kdump;')
+            self.con.run_command( "sed -i 's/KDUMP_FADUMP=\"no\"/KDUMP_FADUMP=\"yes\"/' /etc/sysconfig/kdump;")
+            self.con.run_command( "touch /etc/sysconfig/kdump; systemctl restart kdump.service; sync", timeout=600)
+        elif self.distro == 'RHEL':
+            self.con.run_command( "which stty && stty cols 300; which stty && stty rows 30")
+            self.con.run_command( "grubby --args=\"fadump=on crashkernel=2G-128G:2048M,128G-:8192M\" --update-kernel=/boot/vmlinuz-`uname -r`")
+            self.con.run_command( "sync; sync;")
+            time.sleep(5)
+        self.cv_SYSTEM.goto_state(OpSystemState.OFF)
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        res = self.con.run_command( "systemctl status kdump.service | grep active")
+        if 'exited' not in res[0].strip():
+            print("Fadump service is not configured properly")
+            
+class KernelCrash_Kdump(OpTestKernelBase):
+
```

Comment:
Why kdump stuff is coming here. We already have multiple files/classes doing kdump/fadump stuff.